src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -240,7 +240,7 @@ object PruneDeadFields {
             else
               requestedType.rowType.fieldOption(x.rightFieldMapping(f.name)).map(reqF => f.name -> reqF.typ)
           }: _*),
-          globalType = minimal(requestedType.globalType))
+          globalType = minimal(right.typ.globalType))
```

Comment:
unrelated bug. Not _technically_ a bug in master given that we don't support required top-level structs, but since previous builds were generating required empty global structs we got errors.